src/broccoli/glimmer-build.js

```diff
@@ -0,0 +1,173 @@
+const AbstractBuild = require('ember-build-utilities/lib/abstract-build');
+const TypeScriptCompiler = require('broccoli-typescript-compiler').typescript;
+const resolveLocal = require('ember-build-utilities/lib/paths/resolve-local');
+const addonProcessTree = require('ember-build-utilities/lib/addons/process-tree');
+const GlimmerTemplatePrecompiler = require('ember-build-utilities/lib/compilers/glimmer');
+const notifyAddonIncluded = require('ember-build-utilities/lib/addons/notify-included');
+const shouldIncludeAddon = require('ember-build-utilities/lib/addons/should-include');
+const existsSync = require('exists-sync');
+const p = require('ember-cli-preprocess-registry/preprocessors');
+const BroccoliSource = require('broccoli-source');
+const Funnel = require('broccoli-funnel');
+const { WatchedDir } = BroccoliSource;
+const {
+  isType,
+  preprocessTemplates,
+  preprocessMinifyCss,
+  setupRegistry,
+  defaultRegistry
+} = p;
+
+
+const DEFAULT_TS_OPTIONS = {
+  tsconfig: {
+    compilerOptions: {
+      target: "ESNext",
+      module: "es2015",
+      inlineSourceMap: true,
+      inlineSources: true,
+      moduleResolution: "node"
+    },
+    exclude: [
+      'node_modules',
+      '**/*.d.ts'
+    ]
+  }
+};
+
+let DEFAULT_CONFIG = {
+  outputPaths: {
+    app: {
+      html: 'index.html',
+    }
+  },
+  addons: {},
+};
+
+class GlimmerBuild extends AbstractBuild {
+  constructor(defaults, options = {}) {
+    super(defaults);
+    this.options = options;
+    let missingProjectMessage = 'You must pass through the default arguments passed into your ember-cli-build.js file when constructing a new GlimmerApp';
+    if (arguments.length === 0) {
+      throw new Error(missingProjectMessage);
+    }
+
+    if (!defaults.project) {
+      throw new Error(missingProjectMessage);
+    }
+
+    this.env = process.env.EMBER_ENV || 'development';
+    this.project = defaults.project;
+    this.name = this.project.name();
+    this.targets = options.targets;
+    this.registry = options.registry || p.defaultRegistry(this);
+
+    setupRegistry(this);
+    this._notifyAddonIncluded();
+
+    // this.outputPaths = this.buildOutputPaths(options);
+    // this.detectInvalidBlueprint(options);
+  }
+
+  _initOptions() {
+    return
+  }
+
+  _notifyAddonIncluded() {
```

Comment:
This seems like it will be identical in ember and glimmer apps, should it be hoisted to the base class?